onnx/defs/shape_inference.h

```diff
@@ -181,6 +181,50 @@ inline TensorShapeProto::Dimension operator*(TensorShapeProto::Dimension dim1, T
   return result;
 }
 
+template <typename Container>
+inline std::string stringify(const Container& elements) {
+  std::stringstream ss;
+  for (const auto& element : elements) {
+    ss << element << ", ";
+  }
+  return ss.str();
+}
+
+inline std::pair<int, int> getAttributeElementTypeAndLength(
```

Comment:
nit: It would be helpful to have the code in shape_inference.cc, and just the signature in shape_inference.h (As it is, shape_inference.h has become too bloated with code, and we'd like to avoid this at least going forward.)